### PR TITLE
Clarify GitHub App workflow in README

### PR DESCRIPTION
**Pull request by SpecialThing44**
To close https://github.com/scala-steward-org/scala-steward-action/issues/617
Prompted by wanting to use this action for a single repository and not wanting to create an app/manage secrets. 
See testing https://github.com/SpecialThing44/scala-steward-action-test
Github Actions app does not have permission to write to other repos.
Some other minor clarifications that confused me when I read them.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,6 +18,39 @@ A GitHub Action to launch [Scala Steward](https://github.com/scala-steward-org/s
...
+    runs-on: ubuntu-22.04
+    name: Scala Steward
+    steps:
+      - name: Scala Steward
+        uses: scala-steward-org/scala-steward-action@v2
+```
+
+See [Usage](#usage)
+
+Note that you won't be able to use any of `github-app-*`,`repos-file`, or `github-repository` inputs as the GitHub Actions App can only possibly write to the current repository.
```
Comment (alejandrohdezma) [inline README.md:47]:
```suggestion
Note that you won't be able to use any of `github-app-*`, `repos-file`, or `github-repository` inputs as the GitHub Actions App can only possibly write to the current repository.
```

File: README.md [hunk 2]
```diff
@@ -228,7 +270,8 @@ When it launches it will send PR to update all the repos selected in step (2.2).
     # Location of a `.scala-steward.conf` file with default
     # values.
     # 
-    # If the provided file is missing the action will fail.
+    # If you specify a file and it does not exist in the
```
Comment (alejandrohdezma) [inline README.md:273]: This needs to be changed in [`action.yml`](https://github.com/scala-steward-org/scala-steward-action/blob/master/action.yml) as it is generated from there.

File: README.md [hunk 3]
```diff
@@ -115,6 +151,12 @@ When it launches it will send PR to update all the repos selected in step (2.2).
 
 </details>
 
+## Configuration
+
+There is a `repo-config` option in usage which is a path to a configuration file. It defaults to `.github/.scala-steward.conf`.
+
+See the [configuration documentation](https://github.com/scala-steward-org/scala-steward/blob/main/docs/repo-specific-configuration.md).
+
```
Comment (alejandrohdezma) [inline README.md:159]: I believe this is redundant, as it is already specified in the `Usage` section.

```suggestion
```

**Review by alejandrohdezma — COMMENTED** [review]
Thank you very much @SpecialThing44! 😄 It is great! Left some minor comments or corrections.

File: README.md (hunk 1 above)
Comment (SpecialThing44) [reply README.md]: Fixed!

File: README.md (hunk 2 above)
Comment (SpecialThing44) [reply README.md:273]: Added.

File: README.md (hunk 3 above)
Comment (SpecialThing44) [reply README.md:159]: I don't mind leaving it only in the usage section, but I'm going to add the link in there then, as it is in a separate repo.

File: README.md [hunk 4]
```diff
@@ -99,7 +134,7 @@ jobs:
     steps:
       - name: Launch Scala Steward
         uses: scala-steward-org/scala-steward-action@v2
-        with:
+        with:git
```
Comment (alejandrohdezma) [inline README.md:137]:
```suggestion
        with:
```

**Review by alejandrohdezma — APPROVED** [review]

**Pull request merged**